java/ql/src/Security/CWE/CWE-319/HttpsUrls.ql

```diff
@@ -29,18 +30,6 @@ class HTTPString extends StringLiteral {
   }
 }
 
-class URLConstructor extends ClassInstanceExpr {
```

Comment:
When moving this it has been renamed to `UrlConstructor` so presumably something breaks here?

Reply:
creating a new Pr to address the same.

Reply:
Sorry, I just checked there are uses of this in `OpenStream.ql` and `InsecureBasicAuth.ql` files but they both define their own `URLConstructor` and use that so we are good here.

Reply:
So that just means that there are more opportunities for this refactor? In any case, my comment still applies: This code looks broken, if you try to run this query I expect a compilation failure (you've renamed `URL` to `Url`, which I think is fine, but does require some further updates in this file).

Reply:
@aschackmull The `OpenStream.ql` query is obsoluted by this one. It only include the `OpenStream` call as a sink which I have already included here. There no reason to keep the `OpenStream` PR once this one is merged. As for the `InsecureBasicAuth.ql` file, `@lucha-bc` defines his own set of classes to model URL's. once this Pr is merged, I would open a new PR to remove the redundant code. As for the tests, I can confirm the test suite works fine with the latest set of changes.